### PR TITLE
[Sprint Maintain]Feature Logic Statistic Of Personal

### PR DESCRIPTION
**Pull request by nguyenhuy95dn**
Feature Logic Statistic Of Personal
https://dev.framgia.com/redmine/issues/131118
Effect: N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfChart.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    public List<Chart.DataChart> getNumberLeaveTimes() {
+        return mNumberLeaveTimes;
+    }
+
+    public void setNumberLeaveTimes(List<Chart.DataChart> numberLeaveTimes) {
+        mNumberLeaveTimes = numberLeaveTimes;
+    }
+
+    public class Chart {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfChart.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfChart.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+        public DataChart getDataChart() {
+            return mDataChart;
+        }
+
+        public void setDataChart(DataChart dataChart) {
+            mDataChart = dataChart;
+        }
+
+        public class DataChart {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfChart.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+    public RemainingDaysOff getRemainingDaysOff() {
+        return mRemainingDaysOff;
+    }
+
+    public void setRemainingDaysOff(RemainingDaysOff remainingDaysOff) {
+        mRemainingDaysOff = remainingDaysOff;
+    }
+
+    public class LeaveTime {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+        public int getApprovedInMonth() {
+            return mApprovedInMonth;
+        }
+
+        public void setApprovedInMonth(int approvedInMonth) {
+            mApprovedInMonth = approvedInMonth;
+        }
+    }
+
+    public class DaysOff {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+        public int getSalaryPayByInsuranceInYear() {
+            return mSalaryPayByInsuranceInYear;
+        }
+
+        public void setSalaryPayByInsuranceInYear(int salaryPayByInsuranceInYear) {
+            mSalaryPayByInsuranceInYear = salaryPayByInsuranceInYear;
+        }
+    }
+
+    public class Ot {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+        public int getRejectedInYear() {
+            return mRejectedInYear;
+        }
+
+        public void setRejectedInYear(int rejectedInYear) {
+            mRejectedInYear = rejectedInYear;
+        }
+    }
+
+    public class RemainingDaysOff {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/source/StatisticsDataSource.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import com.framgia.wsm.data.source.remote.api.response.StatisticsResponse;
+import io.reactivex.Single;
+
+/**
+ * Created by nguyenhuy95dn on 8/4/2017.
+ */
+
+public class StatisticsDataSource {
+
+    public interface RemoteDataSource {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/source/StatisticsDataSource.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java (hunk 5 above)
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: WARNING: [ShortClassName] Avoid short class names like Ot
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+        public int getSalaryPayByInsuranceInYear() {
+            return mSalaryPayByInsuranceInYear;
+        }
+
+        public void setSalaryPayByInsuranceInYear(int salaryPayByInsuranceInYear) {
+            mSalaryPayByInsuranceInYear = salaryPayByInsuranceInYear;
+        }
+    }
+
+    public class OTHour {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfChart.java [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+    public List<Chart.DataChart> getNumberLeaveTimes() {
+        return mNumberLeaveTimes;
+    }
+
+    public void setNumberLeaveTimes(List<Chart.DataChart> numberLeaveTimes) {
+        mNumberLeaveTimes = numberLeaveTimes;
+    }
+
+    //Chart
+    public class Chart {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfChart.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfChart.java [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+        public DataChart getDataChart() {
+            return mDataChart;
+        }
+
+        public void setDataChart(DataChart dataChart) {
+            mDataChart = dataChart;
+        }
+
+        //DataChart
+        public class DataChart {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfChart.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+    public RemainingDaysOff getRemainingDaysOff() {
+        return mRemainingDaysOff;
+    }
+
+    public void setRemainingDaysOff(RemainingDaysOff remainingDaysOff) {
+        mRemainingDaysOff = remainingDaysOff;
+    }
+
+    //LeaveTime
+    public class LeaveTime {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+            return mApprovedInMonth;
+        }
+
+        public void setApprovedInMonth(int approvedInMonth) {
+            mApprovedInMonth = approvedInMonth;
+        }
+    }
+
+    //DaysOff
+    public class DaysOff {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+            return mSalaryPayByInsuranceInYear;
+        }
+
+        public void setSalaryPayByInsuranceInYear(int salaryPayByInsuranceInYear) {
+            mSalaryPayByInsuranceInYear = salaryPayByInsuranceInYear;
+        }
+    }
+
+    //OTHour
+    public class OTHour {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 14]
```diff
@@ -0,0 +1,267 @@
...
+            return mRejectedInYear;
+        }
+
+        public void setRejectedInYear(int rejectedInYear) {
+            mRejectedInYear = rejectedInYear;
+        }
+    }
+
+    //RemainingDaysOff
+    public class RemainingDaysOff {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/source/StatisticsDataSource.java [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+import io.reactivex.Single;
+
+/**
+ * Created by nguyenhuy95dn on 8/4/2017.
+ */
+
+public class StatisticsDataSource {
+
+    //RemoteDataSource
+    public interface RemoteDataSource {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/source/StatisticsDataSource.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/data/model/Chart.java [hunk 16]
```diff
@@ -0,0 +1,34 @@
+package com.framgia.wsm.data.model;
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import java.util.ArrayList;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/Chart.java]: ERROR: Unused import: java.util.ArrayList. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/Chart.java]: WARNING: [UnusedImports] Avoid unused imports such as 'java.util.ArrayList'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 17]
```diff
@@ -0,0 +1,299 @@
...
+    }
+
+    public void setRemainingDaysOff(RemainingDaysOff remainingDaysOff) {
+        mRemainingDaysOff = remainingDaysOff;
+    }
+
+    /**
+     * LeaveTime
+     */
+    public class LeaveTime {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: ERROR: [PMD-CPD] 51 lines duplicated.
app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 18]
```diff
@@ -0,0 +1,299 @@
...
+
+        public void setSalaryPayByInsuranceInYear(double salaryPayByInsuranceInYear) {
+            mSalaryPayByInsuranceInYear = salaryPayByInsuranceInYear;
+        }
+    }
+
+    /**
+     * OTHour
+     */
+    public class OTHour {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: ERROR: [PMD-CPD] 51 lines duplicated.
app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java

File: app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java [hunk 19]
```diff
@@ -0,0 +1,249 @@
...
+public class StatisticOfPersonal {
+    @Expose
+    @SerializedName("number_leave_times")
+    private LeaveOverTime mLeaveTimes;
+    @Expose
+    @SerializedName("number_days_off")
+    private DaysOff mDaysOff;
+    @Expose
+    @SerializedName("number_ot_hours")
+    private LeaveOverTime mOtHour;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/data/model/StatisticOfPersonal.java]: mOtHour -> ``mOverTime``

**Pull request merged**